packages/python/plotly/plotly/express/_core.py

```diff
@@ -1082,22 +1106,34 @@ def _isinstance_listlike(x):
 
 
 def _escape_col_name(columns, col_name, extra):
-    while columns is not None and (col_name in columns or col_name in extra):
+    if columns is None:
+        return col_name
+    while col_name in columns or col_name in extra:
         col_name = "_" + col_name
     return col_name
 
 
-def to_unindexed_series(x, name=None):
-    """
-    assuming x is list-like or even an existing pd.Series, return a new pd.Series with
-    no index, without extracting the data from an existing Series via numpy, which
-    seems to mangle datetime columns. Stripping the index from existing pd.Series is
-    required to get things to match up right in the new DataFrame we're building
+def to_unindexed_series(x, name=None, native_namespace=None):
+    """Assuming x is list-like or even an existing Series, returns a new Series with
+    no index (if pandas-like). Stripping the index from existing pd.Series is
+    required to get things to match up right in the new DataFrame we're building.
     """
-    return pd.Series(x, name=name).reset_index(drop=True)
+    x = nw.from_native(x, series_only=True, strict=False)
+    if isinstance(x, nw.Series):
+        return nw.maybe_reset_index(x).rename(name)
+    else:
+        if native_namespace is not None:
+            return nw.new_series(name=name, values=x, native_namespace=native_namespace)
+        elif (pd := nw.dependencies.get_pandas()) is not None:
+            return nw.new_series(name=name, values=x, native_namespace=pd)
+        else:
+            msg = "Pandas installation is required if no namespace is provided."
+            raise NotImplementedError(msg)
```

Comment:
i think here you should also use `import pandas` in a try-except block, else someone would get this error message if they have pandas installed but not imported